conans/test/functional/command/search_test.py

```diff
@@ -1076,3 +1084,242 @@ class Test(ConanFile):
+        client.run("search missing/1.0@conan/stable#revision:234234234234234234 --revisions "
+                   "-r default", assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'missing/1.0@conan/stable#revision'", client.out)
+
+        # Search for a wrong revision in the remote
+        client.run("upload lib/1.0@conan/stable -c --all")
+        client.run("search lib/1.0@conan/stable#revision:234234234234234234 --revisions "
+                   "-r default", assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'lib/1.0@conan/stable#revision'", client.out)
+
+        # Search the right revision but wrong package
+        client.run("search lib/1.0@conan/stable#bd761686d5c57b31f4cd85fd0329751f:"
+                   "234234234234234234 --revisions -r default", assert_error=True)
+        self.assertIn("ERROR: Package not found: "
+                      "'lib/1.0@conan/stable#bd761686d5c57b31f4cd85fd0329751f:"
+                      "234234234234234234'", client.out)
+
+    def search_revision_fail_if_v1_server_test(self):
+        # V1 server
+        test_server = TestServer(users={"conan": "password"}, server_capabilities=[])
+        servers = {"default": test_server}
+        client = TestClient(servers=servers, users={"default": [("conan", "password")]})
+        client.run("search missing/1.0@conan/stable --revisions -r default", assert_error=True)
+        self.assertIn("ERROR: The remote doesn't support revisions", client.out)
+
+    def test_invalid_references_test(self):
+        client = TestClient()
+        # Local errors
+        client.run("search missing/1.0@conan/stable#revision --revisions", assert_error=True)
+        self.assertIn("Cannot list the revisions of a specific revision", client.out)
+
+        client.run("search missing/1.0@conan/stable:pid#revision --revisions", assert_error=True)
+        self.assertIn("Specify a recipe reference with revision", client.out)
+
+        client.run("search missing/1.0@conan/stable#revision:pid#revision --revisions",
+                   assert_error=True)
+        self.assertIn("Cannot list the revisions of a specific package revision", client.out)
+
+        # Remote errors
+        client.run("search missing/1.0@conan/stable#revision --revisions -r fake",
+                   assert_error=True)
+        self.assertIn("Cannot list the revisions of a specific revision", client.out)
+
+        client.run("search missing/1.0@conan/stable:pid#revision --revisions -r fake",
+                   assert_error=True)
+        self.assertIn("Specify a recipe reference with revision", client.out)
+
+        client.run("search missing/1.0@conan/stable#revision:pid#revision --revisions -r fake",
+                   assert_error=True)
+        self.assertIn("Cannot list the revisions of a specific package revision", client.out)
+
+    def test_not_revisions_enabled_but_usage_of_search(self):
```

Comment:
As this test class requires `CONAN_CLIENT_REVISIONS_ENABLED=1` and `CONAN_API_V2_BLOCKED=0` (at least for `TestClient().revisions` to be True), I think it is better to put these tests in its own class, it is hard for me to understand the one with `"CONAN_API_V2_BLOCKED": "1"` inside this test class.

Reply:
Same as above.